beautiful-racket-lib/br/private/generate-literals.rkt

```diff
@@ -9,13 +9,46 @@
          (or (not (regexp-match #rx"[A-Z]" pat-str)) ; either doesn't contain at least one uppercase letter ...
              (not (equal? (string-upcase pat-str) pat-str)))))) ;...  or doesn't contain all uppercase letters
 
+(define (wildcard-identifier? pat-datum)
+  (and (symbol? pat-datum)
+       (not (literal-identifier? pat-datum))
+       (not (memq pat-datum '(... _)))))
+
 ;; generate literals for any symbols that are not ... or _ and not IN_CAPS
 (define (generate-literals pats)
   (for*/list ([pat-arg (in-list (syntax-flatten pats))]
               [pat-datum (in-value (syntax->datum pat-arg))]
               #:when (literal-identifier? pat-datum))
     pat-arg))
 
+(define (all-...-follow-wildcards pats)
+  (define prev-datum (box #f))
+  (and
+    (for*/and ([pat-arg (in-list (syntax-flatten pats))]
+               [pat-datum (in-value (syntax->datum pat-arg))])
+      ;; OK if there's no previous datum,
```

Comment:
I think the compiler puts `set!`'d variables into a box (or the moral equivalent of a box). If that's true, then I don't think there's performance savings either way.

(This is based on what I remember from talking to Leif)

To me, a box is for saying "this variable is going to be mutated".